### ISSUE TITLE
Missing `$result` with `$site->search()` and Algolia 

### ISSUE THREAD

**Issue by marcus-at-localhost** [distantnative/search-for-kirby#27]
## Describe the bug

Using `$site->search('ptfe')` in the frontend, results in this:

![image](https://user-images.githubusercontent.com/1337223/97810764-8cddd900-1c76-11eb-961a-cb12a181196d.png)

This is the result array, before it throws the error

![image](https://user-images.githubusercontent.com/1337223/97810745-61f38500-1c76-11eb-9bef-8c7398500dee.png)

Building the index works, the search in the backend works and the data in Algolia backend is searchable as well.

I've got an Algolia Free Account for testing right now.

I think, what happens, if I look at the request from the search panel in the backend in Provider.php `filterByCollection()` the `$collection` is `null`.

But I don't understand how it get there for the `$site->search('ptfe')` call.

Edit: Ok, when I use `search('ptfe')` then `$collection` is null and it works as expected.

**Comment by distantnative** [comment]
`$collection` cannot be `null` - otherwise the if-statement above would catch it and the error would be different.

It seems like `$result` has no index `id`. I think Algolia renames it to `objectID` in its response.

If you would change `site/plugins/search/src/models/Provider.php` on line 117 from

```php
return $collection->has($result['id']);
```

to

```php
return $collection->has($result['id'] ?? $result['objectID'] ?? null);
```

Does this solve your problem?

**Comment by marcus-at-localhost** [comment]
It ends here.
![image](https://user-images.githubusercontent.com/1337223/99446271-16e89b80-291f-11eb-823a-722f3695db66.png)

Unfortunately, I can't wrap my mind around it.

As I mentioned `search('ptfe')` works as expected.

**Comment by marcus-at-localhost** [comment]
I just noticed that I get the same error above in the panels when I use a [Page](https://getkirby.com/docs/reference/panel/fields/pages) field and want to search for an entry:

![image](https://user-images.githubusercontent.com/1337223/99714146-85576600-2aa5-11eb-929a-288af4e6a62c.png)

```php
    protected function filterByCollection(array $results, $collection = null): array
    {
        // If no collection exists, return all results
        if ($collection === null) {
            return $results;
        }

        clock($results);

        // Otherwise remove the results that are not
        // part of the collection
        return array_filter($results, function ($result) use ($collection) {
            return $collection->has($result['id'] ?? $result['objectID'] ?? null);
        });
    }
```

![image](https://user-images.githubusercontent.com/1337223/99714325-c18ac680-2aa5-11eb-8d39-1e75e06a0c48.png)

The yellow marked object in the screenshot is the output of `clock($results);`

**Comment by distantnative** [comment]
Ok, I think I spotted my mistake.

Does it work when you replace this line:
https://github.com/distantnative/search-for-kirby/blob/main/src/models/Providers/Algolia.php#L114

with:

```php
        $results['hits'] = $this->filterByCollection($results['hits'], $collection);
```
?

**Comment by distantnative** [comment]
This should be solved with the 1.1.0-rc.1.
Please let me know if not.
